merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSuccessfulConstraintRunsAction.java

```diff
@@ -0,0 +1,66 @@
+package gov.nasa.jpl.aerie.merlin.server.remotes.postgres;
+
+import org.intellij.lang.annotations.Language;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static gov.nasa.jpl.aerie.constraints.json.ConstraintParsers.violationP;
+import static gov.nasa.jpl.aerie.merlin.server.remotes.postgres.PostgresParsers.getJsonColumn;
+
+final class GetSuccessfulConstraintRunsAction implements AutoCloseable {
+  private static final @Language("SQL") String sql = """
+    select
+      cr.constraint_id,
+      cr.simulation_dataset_id,
+      cr.status,
+      cr.violations
+    from constraint_run as cr
+    where cr.status = 'resolved'
+    and cr.constraint_id = any(?)
+    and cr.simulation_dataset_id in (select distinct on (simulation_id) id from simulation_dataset order by simulation_id, id desc)
+  """;
+
+  private final PreparedStatement statement;
+  private final List<Long> constraintIds;
+
+  public GetSuccessfulConstraintRunsAction(final Connection connection, final List<Long> constraintIds) throws SQLException {
+    this.statement = connection.prepareStatement(sql);
+    this.constraintIds = constraintIds;
+  }
+
+  public List<ConstraintRunRecord> get() throws SQLException, ConstraintRunRecord.Status.InvalidRequestStatusException {
+    this.statement.setArray(1, this.statement.getConnection().createArrayOf("integer", constraintIds.toArray()));
+
+    try (final var results = this.statement.executeQuery()) {
+      final var constraintRuns = new ArrayList<ConstraintRunRecord>();
+
+      while (results.next()) {
+        final var constraintId = results.getLong("constraint_id");
+        final var status = ConstraintRunRecord.Status.fromString(results.getString("status"));
```

Comment:
This line needs to be updated to `results.getBoolean("status")`

Reply:
I think you were seeing a weird state, that line of code is gone now!